spec/services/scishield_api_adapter_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ResearchSafetyAdapters::ScishieldApiAdapter do
+  subject(:adapter) { described_class.new(user) }
+  let(:user) { build(:user, email: "research@osu.edu") }
+  let(:api_endpoint) { "#{adapter.client.class::API_ENDPOINT}?#{adapter.client.training_query(user.email)}" }
+
+  describe "with a successful response" do
+    let(:response) { File.expand_path("../../fixtures/scishield/success.json", __dir__) }
+
+    before do
+      stub_request(:get, api_endpoint)
+        .to_return(
+          body: File.new(response),
+          status: 200,
+        )
+    end
+
+    it "is certified for Lab Safety" do
+      certificate = instance_double(ResearchSafetyCertificate, name: "Lab Safety Training for Lab Workers")
+      expect(adapter).to be_certified(certificate)
+    end
+
+    it "is certified for Hazardous Waste" do
+      certificate = instance_double(ResearchSafetyCertificate, name: "Hazardous Waste Awareness Training")
+      expect(adapter).to be_certified(certificate)
+    end
+
+    it "is not certified for Fire Extinguisher Course" do
+      certificate = instance_double(ResearchSafetyCertificate, name: "OSU Fire Extinguisher Course (online) ")
+      expect(adapter).not_to be_certified(certificate)
+    end
+
+    it "only calls the response once" do
+      expect(adapter.client).to receive(:certifications_for).once.and_call_original
+      adapter.certified? instance_double(ResearchSafetyCertificate, name: "Lab Safety Training for Lab Workers")
+      adapter.certified? instance_double(ResearchSafetyCertificate, name: "Hazardous Waste Awareness Training")
+      adapter.certified? instance_double(ResearchSafetyCertificate, name: "RANDOM")
+    end
+  end
+
+  describe "unable to connect" do
+    before do
+      stub_request(:get, api_endpoint)
+        .to_timeout
+    end
+
+    it "raises an error" do
+      expect { adapter.certified?(anything) }.to raise_error(Timeout::Error)
+    end
+  end
+
+  describe "the user is not found" do
+    let(:response) { File.expand_path("../../fixtures/scishield/user_not_found.json", __dir__) }
+    before do
+      stub_request(:get, api_endpoint)
+        .to_return(
+          body: File.new(response),
+          status: 200,
+        )
+    end
+
+    it "doesn't raise an error, and doesn't certify anything" do
+      random = instance_double(ResearchSafetyCertificate, name: "RANDOM")
+      expect(adapter).not_to be_certified(random)
+    end
+  end
+
+  describe "the user is found but the response includes an error" do
+    let(:response) { File.expand_path("../../fixtures/scishield/errors.json", __dir__) }
+    before do
+      stub_request(:get, api_endpoint)
+        .to_return(
+          body: File.new(response),
+          status: 200,
```

Comment:
Should this be `200` or `403`? Looks like this is a 403 Forbidden response, rather than an 200 OK response